<?xml version="1.0" encoding="utf-8"?>
<issues>
  <issue>
    <field name="Priority">
      <value>Major</value>
    </field>
    <field name="Type">
      <value>Bug</value>
    </field>
    <field name="State">
      <value>Incomplete</value>
    </field>
    <field name="Assignee" />
    <field name="Subsystem">
      <value>Framework</value>
    </field>
    <field name="Fix versions" />
    <field name="Affected versions" />
    <field name="Fixed in build" />
    <field name="numberInProject">
      <value>159</value>
    </field>
    <field name="summary">
      <value>Exception should be thrown when an insert fails</value>
    </field>
    <field name="description">
      <value>As per discussion on castle users mailing list titled "Should an exception be thrown when an insert fails".

Tim: I'd accidentally missed marking a model property with the property attribute on a required field.  This meant it wasn't being included in the insert SQL statement, so the the insert was failing at the DB level.  This wasn't being reported as an error in Active Record though.

Leonard: I can get this on any of my ActiveRecord projects simply by changing the name of one of my properties so it doesn't match the database field and trying to save.

David: This is a side effect of ActiveRecord flushing when it has had an error - the original error gets lost because of the new error that occurs during the flush. I put in some code to fix this in the patch I created to make ActiveRecord use the validator component.

</value>
    </field>
    <field name="created">
      <value>1180251989000</value>
    </field>
    <field name="updated">
      <value>1181092978000</value>
    </field>
    <field name="reporterName">
      <value>tim</value>
    </field>
    <field name="updaterName">
      <value>tim</value>
    </field>
    <field name="resolved">
      <value>1282136353229</value>
    </field>
    <field name="permittedGroup">
      <value>All Users</value>
    </field>
  </issue>
</issues>